File: backend/data-service/src/main/java/com/padd/data/entity/HealthRecord.java
package com.padd.data.entity;

import java.time.LocalDateTime;

import jakarta.persistence.*;

@Entity
@Table(name = "health_record")
public class HealthRecord {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "timestamp", columnDefinition = "TIMESTAMP")
    private LocalDateTime timestamp;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "blood_pressure_id", referencedColumnName = "id")
    private BloodPressure bloodPressure;

    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    private double heartRate;
    private int stressLevel;
    private double bloodOxygenation;
    private double bodyTemperature;
    private double acceleration;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(LocalDateTime timestamp) {
        this.timestamp = timestamp;
    }

    public BloodPressure getBloodPressure() {
        return bloodPressure;
    }

    public void setBloodPressure(BloodPressure bloodPressure) {
        this.bloodPressure = bloodPressure;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public double getHeartRate() {
        return heartRate;
    }

    public void setHeartRate(double heartRate) {
        this.heartRate = heartRate;
    }

    public int getStressLevel() {
        return stressLevel;
    }

    public void setStressLevel(int stressLevel) {
        this.stressLevel = stressLevel;
    }

    public double getBloodOxygenation() {
        return bloodOxygenation;
    }

    public void setBloodOxygenation(double bloodOxygenation) {
        this.bloodOxygenation = bloodOxygenation;
    }

    public double getBodyTemperature() {
        return bodyTemperature;
    }

    public void setBodyTemperature(double bodyTemperature) {
        this.bodyTemperature = bodyTemperature;
    }

    public double getAcceleration() {
        return acceleration;
    }

    public void setAcceleration(double acceleration) {
        this.acceleration = acceleration;
    }
    
}

File: backend/user-service/app/models/health_record.ts
import { DateTime } from 'luxon'
import { BaseModel, column, belongsTo, hasOne } from '@adonisjs/lucid/orm'
import type { BelongsTo, HasOne } from '@adonisjs/lucid/types/relations'
import BloodPressure from './blood_pressure.js'
import User from './user.js'

export default class HealthRecord extends BaseModel {
  static table = 'health_record'

  @column({ isPrimary: true })
  declare id: number

  @column.dateTime({ autoCreate: true })
  declare timestamp: DateTime

  @column()
  declare bloodPressureId: number

  @hasOne(() => BloodPressure, {
    foreignKey: 'id',
    localKey: 'bloodPressureId',
  })
  declare bloodPressure: HasOne<typeof BloodPressure>

  @column()
  declare userId: number

  @belongsTo(() => User, {
    foreignKey: 'userId',
  })
  declare user: BelongsTo<typeof User>

  @column()
  declare heartRate: number

  @column()
  declare stressLevel: number

  @column()
  declare bloodOxygenation: number

  @column()
  declare bodyTemperature: number

  @column()
  declare acceleration: number
}
